parser/src/lexer.rs

```diff
@@ -29,6 +29,11 @@ impl<I> StreamOnce for LocatedStream<I>
             .uncons()
             .map(|ch| {
                 self.location.bump(ch);
+                // HACK: The layout algorithm expects `1` indexing for columns -
+                // this could be altered in the future though
+                if self.location.column == Column::from(0) {
+                    self.location.column = Column::from(1);
+                }
```

Comment:
Should this not add `+1` instead?

Might be an issue if the indentation is only a by a single space since the first and second column will look like they are the same.
```f#
f 1
 2
// Is parsed like
f 1
2
// instead of
f 1 2
```